keps/sig-node/127-user-namespaces/README.md

```diff
@@ -316,121 +327,108 @@ The picked range will be stored under a file named `userns` in the pod folder
-supported volume type to handle the fsUser new param, and the corresponding
-volume SetUpAt() function to pass this new param when calling AtomicWriter.
+This means user 0 inside the container is mapped to user 65536 on the host. Let'
+say the pod is running UID 0 inside the container too.
 
-We have written already this code and the per-volume diff is 3 lines.
+##### Example without idmap mounts
 
-For fsGroup() we don't need any changes, these volume types already honors it.
+To understand how idmap mounts work, let's first see an example of what happens
+without idmap mounts.
 
-[volume-mounter-args]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/volume.go#L125-L129
-[operation-executor]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/util/operationexecutor/operation_generator.go#L674-L675
-[volumeHost interface]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/plugins.go#L360-L361
-[atomic-writer-fsUser]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/util/atomic_writer.go#L68
-[configmap-fsuser]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/configmap/configmap.go#L272-L273
+If such a pod creates a file in any volume, as its mapped to UID 65536 on the
+host, the files it creates will be owned by UID 65536 (i.e. the on disk inode
+will have that value). The pod can read and write to this file just fine. When
+reading the file, the kernel translates the UID 65536 to UID 0 inside the
+container, which is the UID we are running as, and therefore we can read the
+file
 
-#### Phase 2: pods with volumes
+Problems arise if the pod is re-scheduled with another mapping. Let's say it is
+scheduled now with this mapping:
 
-This phase makes user namespaces work in pods with volumes too. This is
-activated via the bool `pod.spec.HostUsers` too and pods fall into this mode if
-some other volume type than the ones listed for phase 1 is used. IOW, when phase
-2 is implemented, pods that use volume types not supported in phase 1, fall into
-the phase 2.
+| Host ID  | Container ID   | Length  |
+|----------|----------------|---------|
+| 131072   |      0         |  65536  |
 
-All pods in this mode will use _the same_ mapping, chosen by the kubelet, with a
-length 65536, and mapping the range 0-65535 too. IOW, each pod will have its own user
-namespace, but they will map to _the same_ UIDs/GIDs in the host.
 
-Using the same mapping allows for pods to share files and mitigates all the
-listed vulnerabilities (as the host is protected from the container). It is also
-a trivial next-step to take, given that we have phase 1 implemented: just return
-the same mapping if the pod has other volumes.
+Now, when trying to read the file the pod created with UID 65536 on its on disk
```

Comment:
Since this KEP now *only* supports stateless pods, how would this situation ever arise?